File: chapters/01_gateway_event_loop/v1_sync_flow.ts
// TypeScript mirror for Chapter 01: Gateway Event Loop
// Run with: npx ts-node v1_sync_flow.ts

function decideAction(text: string): [string, string] {
  const lowered = text.toLowerCase();
  if (lowered.includes("create") || text.includes("创建")) {
    return ["create_file", "demo.txt"];
  }
  if (lowered.includes("read") || text.includes("读取")) {
    return ["read_file", "demo.txt"];
  }
  return ["reply", "-"];
}

function handleSync(text: string): Record<string, string> {
  const [action, target] = decideAction(text);
  let egress: string;

  if (action === "create_file") {
    egress = `created:${target}`;
  } else if (action === "read_file") {
    egress = `content:${target}=<mock>`;
  } else {
    egress = `reply:${text}`;
  }

  return {
    ingress: text,
    action,
    target,
    egress,
  };
}

function main(): void {
  const out = handleSync("请创建一个演示文件");
  console.log(out);
}

main();

File: chapters/01_gateway_event_loop/v1_sync_flow.py
from __future__ import annotations


def decide_action(text: str) -> tuple[str, str]:
    lowered = text.lower()
    if "create" in lowered or "创建" in text:
        return "create_file", "demo.txt"
    if "read" in lowered or "读取" in text:
        return "read_file", "demo.txt"
    return "reply", "-"


def handle_sync(text: str) -> dict[str, str]:
    action, target = decide_action(text)
    if action == "create_file":
        egress = f"created:{target}"
    elif action == "read_file":
        egress = f"content:{target}=<mock>"
    else:
        egress = f"reply:{text}"

    return {
        "ingress": text,
        "action": action,
        "target": target,
        "egress": egress,
    }


def main() -> None:
    out = handle_sync("请创建一个演示文件")
    print(out)


if __name__ == "__main__":
    main()
